File: modules/coupon-crawler-service/src/main/java/com/thanh0x/coursedeal/CouponCrawlerServiceApplication.java
package com.thanh0x.coursedeal;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.ComponentScan;

@SpringBootApplication
@ComponentScan(basePackages = "com.thanh0x.coursedeal")
public class CouponCrawlerServiceApplication {
    public static void main(String[] args) {
        SpringApplication.run(CouponCrawlerServiceApplication.class, args);
    }
}

File: modules/coupon-crawler-service/src/main/kotlin/com/thanh0x/coursedeal/CouponCrawlerServiceApplication.kt
package com.thanh0x.coursedeal

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.ComponentScan

@SpringBootApplication
@ComponentScan(basePackages = ["com.thanh0x.coursedeal"])
class CouponCrawlerServiceApplication

@Suppress("SpreadOperator")
fun main(args: Array<String>) {
    // runApplication's vararg signature requires spreading the incoming args array.
    runApplication<CouponCrawlerServiceApplication>(*args)
}
